src/de/id.rs

```diff
@@ -1,6 +1,8 @@
 use serde::de::{self, Visitor};
 
-use super::{Deserializer, Error, Result};
+use super::{Deserializer, ErrorCode};
+
+type Result<T> = std::result::Result<T, ErrorCode>;
```

Comment:
perhaps another typedef for `Result<T, Error>` would help here?

Reply:
I've added a second typedef, now for the naming. I've remapped `ErrorCode` => `Error` as this is now the error type returned by both the `Serializer` and `Deserializer`, `type Result<T, E = Error> = std::result::Result<T, E>` defaults to this error type as well but allows importing the `ron` `Result` without losing the functionality of `std::result::Result`. Furthermore, `Error` => `SpannedError` and `type SpannedResult<T> = std::result::Result<T, SpannedError>`, which is returned by all convenience function around deserialization. I'm not sure if `SpannedError` is the best name, `PositionalError` could also work. This renaming (well, this PR in general) will break some code at the benefit of clarifying that only deserialization errors can have nice error position information + they will now always have this.